src/strategies/ekubo_base.py

```diff
@@ -0,0 +1,60 @@
+import math
+from src.utils.ekubo_math import price_to_tick, impermanent_loss
+
+class EkuboStrategyBase:
+    def run_simulation(self, liquidity_range, rebalance_range, train_swap_df, initial_liquidity=10000000, fee_percentage=0.00001, decimal = 18):
+        """
+        Run a simulation based on the provided swap data and strategy parameters.
+
+        Parameters:
+        - liquidity_range: The range within which liquidity is considered active.
+        - rebalance_range: The range within which rebalancing is not triggered.
+        - train_swap_df: DataFrame containing the swap events to simulate.
+        - initial_liquidity: Initial liquidity provided in the pool.
+        - fee_percentage: The fee percentage for swaps.
+
+        Returns:
+        - net_earnings: Net earnings after considering fees and impermanent loss.
+        - fees_earned: Total fees earned during the simulation.
+        - impermanent_loss_total: Total impermanent loss during the simulation.
+        """
+        # Check if train_swap_df is empty
+        if train_swap_df.empty:
+            raise ValueError("train_swap_df is empty, cannot run simulation.")
+
+        # Ensure the price column exists
+        if 'price' not in train_swap_df.columns:
+            raise ValueError("The 'price' column is missing in train_swap_df.")
+
+        initial_price = train_swap_df['price'].iloc[0]
+        fees_earned = 0
+        impermanent_loss_total = 0
+        current_liquidity = initial_liquidity
+
+        for swap_event in train_swap_df.itertuples():
+            current_price = swap_event.price
+            amount_swapped = swap_event.amount1
+            available_liquidity = swap_event.liquidity_after
```

Comment:
Add a note that this is not accurate way to compute. We will check final results and if they match with expected values, we can accept this, else we may have to look deep. 

Reason for inaccuracy:
- during a swap, price can move from one tick to another (can u guess why?)
- the liquidity after only represents final tick and not the ticks the price has moved in between, this leads to inaccurate fee compute
- but this is not common event, happens when someone is selling a lot. but can happen.